app/controllers/api/generic_events_controller.rb

```diff
@@ -30,6 +30,7 @@ def event_attributes
 
         attributes.merge!('supplier' => doorkeeper_application_owner) if doorkeeper_application_owner
         attributes.merge!('eventable' => eventable) if eventable_params
+        attributes.merge!('created_by' => created_by) if created_by
```

Comment:
Uses the api controller set `created_by`, so no need for the deleted method below.